platforms/opencl/src/OpenCLNonbondedUtilities.cpp

```diff
@@ -354,7 +354,13 @@ void OpenCLNonbondedUtilities::prepareInteractions(int forceGroups) {
     context.executeKernel(kernels.findInteractingBlocksKernel, context.getNumAtoms(), interactingBlocksThreadBlockSize);
     forceRebuildNeighborList = false;
     lastCutoff = kernels.cutoffDistance;
-    context.getQueue().enqueueReadBuffer(interactionCount.getDeviceBuffer(), CL_FALSE, 0, sizeof(int), pinnedCountMemory, NULL, &downloadCountEvent); 
+    context.getQueue().enqueueReadBuffer(interactionCount.getDeviceBuffer(), CL_FALSE, 0, sizeof(int), pinnedCountMemory, NULL, &downloadCountEvent);
+
+    #if __APPLE__ && defined(__aarch64__)
+    // Segment the command stream to avoid stalls later.
+    if (groupKernels[forceGroups].hasForces && useCutoff && numTiles > 0)
```

Comment:
It already checks for some of these conditions above:

```c++
    if (!useCutoff)
        return;
    if (numTiles == 0)
        return;
```

Now I'm confused about why it doesn't also return if `hasForces` is false, and likewise why the call to `updateNeighborListSize()` in `computeInteractions()` isn't inside the `if (kernels.hasForces)` block.  Is there ever a reason it would need to rebuild the neighbor list if there are no nonbonded forces in the specified force groups?  I can think of any.  I think we probably want to make those changes, though we need to make sure all the tests pass in case there's some reason I've forgotten about.

Reply:
I think that would belong in a separate PR. If changing that causes a bug, I don't want this optimization to be conflated with it and unnecessarily reverted.

Reply:
Ok, that's fine.